### PR TITLE
Afterpay payment method country available. 

### PR DESCRIPTION
**Pull request by tiutiunnykleasoft**
Afterpay payment method country available. Rename IPBlockSubscriber.php -> PaymentKeeperSubscriber.php

### CODE REVIEW

File: custom/plugins/emspay/Resources/config.xml [hunk 1]
```diff
@@ -56,12 +64,16 @@
...
-            <description lang="fr">Vous pouvez spécifier des adresses IP spécifiques pour lesquelles Klarna Pay Later est visible, par exemple si vous souhaitez tester Klarna Pay Later. Vous pouvez taper des adresses IP comme 128.0.0.1, 255.255.255.255. Si vous laissez ce Klarna Pay Later vide, il est visible par toutes les adresses IP. "</description>
+            <name>emsonline_afterpay_countries</name>
+            <label>Countries available for AfterPay</label>
+            <label lang="de">Für AfterPay verfügbare Länder</label>
+            <label lang="nl">Landen beschikbaar voor AfterPay</label>
+            <label lang="fr">Pays disponibles pour AfterPay</label>
+            <value>NL, BE</value>
+            <description lang="en">To allow AfterPay to be used for any other country just add its country code (in ISO 2 standard) to the "Countries available for AfterPay" field. Example: BE, NL, FR.</description>
```
Comment (LeasoftKonstantinShapoval) [inline custom/plugins/emspay/Resources/config.xml]: Needs add description for empty values:
EN "<br> If field is empty then AfterPay will be available for all countries."
DE "<br> Wenn das Feld leer ist, ist AfterPay fur alle Lander verfugbar."
FR "<br> Si le champ est vide, AfterPay sera disponible pour tous les pays."
NL "<br> Als het veld leeg is, is AfterPay beschikbaar voor alle landen."

Comment (tiutiunnykleasoft) [reply custom/plugins/emspay/Resources/config.xml]: Thanks for localized answer,

**Review by LeasoftKonstantinShapoval — APPROVED** [review]

File: custom/plugins/emspay/Subscriber/PaymentKeeperSubscriber.php [hunk 2]
```diff
@@ -1,56 +1,74 @@
...
+
+    /**
+     * Check that the user's country matches the country for which Afterpay payment method is available
+     * @param $method
+     * @param $country
+     * @return bool
+     */
+
+    private function countryValidation($method,$country){
+        $countries_string = $this->config[implode('_',['emsonline',$method,'countries'])];
```
Comment (volgustleasoft) [inline custom/plugins/emspay/Subscriber/PaymentKeeperSubscriber.php]: Could it be that we are accessing an array element by a key that does not exist? For example: $this->config['emsonline_ideal_countries']?

File: custom/plugins/emspay/Subscriber/PaymentKeeperSubscriber.php [hunk 3]
```diff
@@ -1,56 +1,74 @@
...
+        return in_array($country, $countries_list);
+    }
+
+    /**
+     * Check current IP address with Test IP Addresses
+     * @param $method
+     * @return bool
+     */
+    private function ipAddressValidation($method){
+        $test_ip = $this->config[implode('_',['emsonline_test_ip',$method])];
```
Comment (volgustleasoft) [inline custom/plugins/emspay/Subscriber/PaymentKeeperSubscriber.php]: Could it be that we are accessing an array element by a key that does not exist? For example: $this->config['emsonline_test_ip_ideal'] ?

Comment (tiutiunnykleasoft) [reply custom/plugins/emspay/Subscriber/PaymentKeeperSubscriber.php]: there is all okay I have a check later

File: custom/plugins/emspay/Subscriber/PaymentKeeperSubscriber.php (hunk 2 above)
Comment (tiutiunnykleasoft) [reply custom/plugins/emspay/Subscriber/PaymentKeeperSubscriber.php]: there is all okay I have a check later

**Review by volgustleasoft — APPROVED** [review]

**Pull request merged**